File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        s_string = {}
        t_string = {}

        for i in range(len(s)):
            if s[i] not in s_string:
                s_string[s[i]] = i

            if t[i] not in t_string:
                t_string[t[i]] = i

            if s_string[s[i]] != t_string[t[i]]:
                return False

        return True               

File: 0205-isomorphic-strings/0205-isomorphic-strings.dart
class Solution {
  bool isIsomorphic(String s, String t) {
    Map<String, String> mapS = {};
    Map<String, String> mapT = {};

    for(int i = 0 ; i < s.length ; i++) {
        String si = s[i];
        String ti = t[i];

        if(mapS.containsKey(si)) {
            if(mapS[si] != ti) return false;
        } else {
            if(mapT.containsKey(ti)) {
                if(mapT[ti] != si) return false;
            }
        }

        mapS[si] = ti;
        mapT[ti] = si;
    }
    return true;
  }
}